lib/backend/k8s/k8s_test.go

```diff
@@ -1967,6 +1987,227 @@ var _ = testutils.E2eDatastoreDescribe("Test Watch support", testutils.Datastore
 		ctx = context.Background()
 	})
 
+	Describe("watching Profiles", func() {
+		createTestServiceAccount := func(name string) {
+			sa := k8sapi.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+			}
+			_, err := c.ClientSet.CoreV1().ServiceAccounts("default").Create(&sa)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		deleteAllServiceAccounts := func() {
+			var zero int64
+			err := c.ClientSet.CoreV1().ServiceAccounts("default").DeleteCollection(&metav1.DeleteOptions{GracePeriodSeconds: &zero}, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+		BeforeEach(func() {
+			createTestServiceAccount("test-sa-1")
+			createTestServiceAccount("test-sa-2")
+		})
+		AfterEach(func() {
+			deleteAllServiceAccounts()
+		})
+		It("supports watching a specific profile (from namespace)", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "kns.default", Kind: apiv3.KindProfile}, "")
+			Expect(err).NotTo(HaveOccurred())
+			event := ExpectAddedEvent(watch.ResultChan())
+			Expect(event.New.Key.String()).To(Equal("Profile(kns.default)"))
+		})
+		It("supports watching a specific profile (from serviceAccount)", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "ksa.default.test-sa-1", Kind: apiv3.KindProfile}, "")
+			Expect(err).NotTo(HaveOccurred())
+			event := ExpectAddedEvent(watch.ResultChan())
+			Expect(event.New.Key.String()).To(Equal("Profile(ksa.default.test-sa-1)"))
+		})
+		It("supports watching all profiles", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Kind: apiv3.KindProfile}, "")
+			Expect(err).NotTo(HaveOccurred())
+			ExpectAddedEvent(watch.ResultChan())
+		})
+		It("rejects names without prefixes", func() {
+			_, err := c.Watch(ctx, model.ResourceListOptions{Name: "default", Kind: apiv3.KindProfile}, "")
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("Unsupported prefix for resource name: default"))
+		})
+	})
+
+	Describe("watching NetworkPolicies (native)", func() {
+		createTestNetworkPolicy := func(name string) {
+			np := networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+			}
+			_, err := c.ClientSet.NetworkingV1().NetworkPolicies("default").Create(&np)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		deleteAllNetworkPolicies := func() {
+			var zero int64
+			err := c.ClientSet.NetworkingV1().NetworkPolicies("default").DeleteCollection(&metav1.DeleteOptions{GracePeriodSeconds: &zero}, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+		BeforeEach(func() {
+			createTestNetworkPolicy("test-net-policy-1")
+			createTestNetworkPolicy("test-net-policy-2")
+		})
+		AfterEach(func() {
+			deleteAllNetworkPolicies()
+		})
+		It("supports watching a specific networkpolicy", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "test-net-policy-1", Kind: apiv3.KindNetworkPolicy}, "")
```

Comment:
Grr, I think we missed this one too - note the mismatch between the name we're passing in and the name we're receiving on the watch. They should be the same, just like for Profiles..